trie.c

```diff
@@ -65,12 +75,12 @@ bool st_find_longest_chain(st_trie_t *trie, st_key_buffer_t *search, st_trie_pay
 	if (code & TRIE_MATCH_BIT) {
         // match nodes are side attachments, so decrease depth
         depth--;
+        // record this as the new longest match
+        longest_match->trie_match_index = offset;
+        longest_match->context_match_len = depth + 1;
         // If bit 14 is also set, there is a child node after the completion string
-        if ((code & TRIE_BRANCH_BIT) && st_find_longest_chain(trie, search, res, offset+2, depth+1))
+        if ((code & TRIE_BRANCH_BIT) && st_find_longest_chain(trie, search, longest_match, offset+2, depth+1))
             return true;
-        // If no better match found deeper, so recordd the payload result!
-        st_get_payload_from_code(res, code, TDATA(offset + 1));
-        res->context_match_len = depth + 1;
 #ifdef SEQUENCE_TRANSFORM_TRIE_SANITY_CHECKS
```

Comment:
remove this sanity check since it no longer builds, and we will do this somewhere else anyways.